### DISCUSSION TITLE
I2P transport ?

### DISCUSSION THREAD

**Discussion by ple1n** [libp2p/rust-libp2p#3105] [General]
or onion routing.

**Comment by Mubelotix** [comment]
This has been discussed [here](https://github.com/libp2p/rust-libp2p/pull/2899#issuecomment-1297231845) and as of today, it appears that support for this kind of transports should be implemented and maintained by the community. It's totally doable in theory

**Reply by umgefahren** [reply]
Once #2899 is somewhat solved you can use Tor. Feel free to add I2P support once we figured that out.
